apps/dashboard/src/components/add-member-modal.tsx

```diff
@@ -147,7 +147,24 @@ ${memberIds.join("\n")}
     }, [onClose, _memberIds, group, signer])
 
     const generateInviteLink = useCallback(async () => {
-        const inviteLink = await bandadaAPI.generateMagicLink(group.id)
+        let inviteLink = null
+
+        if (group.type === "off-chain") {
+            inviteLink = await bandadaAPI.generateMagicLink(group.id)
+        } else {
+            const associatedGroup = await bandadaAPI.getGroupByName(
+                group.name,
+                "on-chain"
+            )
+
+            if (associatedGroup && associatedGroup.length > 0) {
+                inviteLink = await bandadaAPI.generateMagicLink(
+                    associatedGroup[0].id
+                )
+            } else {
+                alert("No associated on-chain group found")
```

Comment:
What do you think if a new associated group is created when users click on `Generate new link` if there is no associated group? Instead of displaying the alert message. 

This way on-chain groups that were created before can benefit from this new feature too.

Reply:
Good idea, I'll add this in.